### PR TITLE
feat: removing singular name functions and changing them to plural naming

### PR DESCRIPTION
**Pull request by MarIniOnz**
Removed functions:
- `add_node`
- `add_edge`

Renamed functions:
- `remove_node` -> `remove_nodes`
- `remove_edge` -> `remove_edges`
- `remove_group` -> `remove_groups`
- `add_node_to_group` -> `add_nodes_to_group`
- `add_edge_to_group` -> `add_edges_to_group`
- `remove_node_from_group` -> `remove_nodes_from_group`
- `remove_edge_from_group` -> `remove_edges_from_group`

**For testing, a `make install-dev` action is required!**

### CODE REVIEW

File: medmodels/medrecord/medrecord.py [hunk 1]
```diff
@@ -566,41 +569,15 @@ def edges_connecting(
...
 
     @overload
-    def remove_node(
+    def remove_nodes(
         self, node: Union[NodeIndexInputList, NodeOperation]
     ) -> Dict[NodeIndex, Attributes]: ...
 
-    def remove_node(
+    def remove_nodes(
         self, node: Union[NodeIndex, NodeIndexInputList, NodeOperation]
```
Comment (LauraBoenchenLB) [inline medmodels/medrecord/medrecord.py:581]: Should we rename the variable to `nodes` ?

File: medmodels/medrecord/medrecord.py [hunk 2]
```diff
@@ -970,7 +906,7 @@ def add_group(
         else:
             return self._medrecord.add_group(group, None, None)
 
-    def remove_group(self, group: Union[Group, GroupInputList]) -> None:
+    def remove_groups(self, group: Union[Group, GroupInputList]) -> None:
```
Comment (LauraBoenchenLB) [inline medmodels/medrecord/medrecord.py:909]: same here, should the variable also be called `groups`

File: medmodels/medrecord/medrecord.py [hunk 3]
```diff
@@ -997,13 +933,13 @@ def add_node_to_group(
...
+            return self._medrecord.add_nodes_to_group(group, self.select_nodes(node))
 
-        return self._medrecord.add_node_to_group(
+        return self._medrecord.add_nodes_to_group(
             group, node if isinstance(node, list) else [node]
         )
 
-    def add_edge_to_group(
+    def add_edges_to_group(
         self, group: Group, edge: Union[EdgeIndex, EdgeIndexInputList, EdgeOperation]
```
Comment (LauraBoenchenLB) [inline medmodels/medrecord/medrecord.py:943]:
```suggestion
        self, group: Group, edges: Union[EdgeIndex, EdgeIndexInputList, EdgeOperation]
```

File: medmodels/medrecord/medrecord.py [hunk 4]
```diff
@@ -979,11 +915,11 @@ def remove_group(self, group: Union[Group, GroupInputList]) -> None:
...
             None
         """
-        return self._medrecord.remove_group(
+        return self._medrecord.remove_groups(
             group if isinstance(group, list) else [group]
         )
 
-    def add_node_to_group(
+    def add_nodes_to_group(
         self, group: Group, node: Union[NodeIndex, NodeIndexInputList, NodeOperation]
```
Comment (LauraBoenchenLB) [inline medmodels/medrecord/medrecord.py:923]:
```suggestion
        self, group: Group, nodes: Union[NodeIndex, NodeIndexInputList, NodeOperation]
```

**Review by LauraBoenchenLB — CHANGES_REQUESTED** [review]
Great job with the renaming and typing! Some thoughts about the variable names, maybe we should also keep these aligned with the function names.

File: medmodels/medrecord/medrecord.py (hunk 1 above)
Comment (MarIniOnz) [reply medmodels/medrecord/medrecord.py:581]: yes

File: medmodels/medrecord/builder.py [hunk 5]
```diff
@@ -53,17 +41,20 @@ def is_node_input(value: object) -> TypeIs[NodeInput]:
...
 
+def is_edge_input(value: object) -> TypeIs[EdgeInput]:
+    return (
+        is_edge_tuple(value)
+        or is_edge_tuple_list(value)
+        or is_pandas_edge_dataframe_input(value)
+        or is_pandas_edge_dataframe_input_list(value)
+        or is_polars_edge_dataframe_input(value)
+        or is_polars_edge_dataframe_input_list(value)
+    )
```
Comment (JabobKrauskopf) [inline medmodels/medrecord/builder.py:55]: If `EdgeInput` is moved, this should also be moved

File: medmodels/medrecord/types.py [hunk 6]
```diff
@@ -87,6 +87,26 @@
...
+    PandasNodeDataFrameInput,
+    List[PandasNodeDataFrameInput],
+    PolarsNodeDataFrameInput,
+    List[PolarsNodeDataFrameInput],
+]
+
+#: A type alias for input to an edge.
+EdgeInput = Union[
+    EdgeTuple,
+    Sequence[EdgeTuple],
```
Comment (JabobKrauskopf) [inline medmodels/medrecord/types.py:103]:
```suggestion
    List[EdgeTuple],
```
as `EdgeTuple` is covariant

File: medmodels/medrecord/types.py [hunk 7]
```diff
@@ -87,6 +87,26 @@
 #: A type alias for input to a Pandas DataFrame for edges.
 PandasEdgeDataFrameInput: TypeAlias = Tuple[pd.DataFrame, str, str]
 
+#: A type alias for input to a node.
+NodeInput = Union[
+    NodeTuple,
+    Sequence[NodeTuple],
```
Comment (JabobKrauskopf) [inline medmodels/medrecord/types.py:93]:
```suggestion
    List[NodeTuple],
```
as `NodeTuple` is covariant

**Review by JabobKrauskopf — CHANGES_REQUESTED** [review]

File: medmodels/medrecord/types.py (hunk 6 above)
Comment (MarIniOnz) [reply medmodels/medrecord/types.py:103]: Created new Tying EdgeInputBuilder and NodeInputBuilder in the builder.py file

File: medmodels/medrecord/medrecord.py [hunk 8]
```diff
@@ -849,6 +783,8 @@ def add_edges(
         ) or is_polars_edge_dataframe_input_list(edges):
             return self.add_edges_polars(edges, group)
         else:
+            if is_edge_tuple(edges):
+                edges = [edges]
             edge_indices = self._medrecord.add_edges(edges)
```
Comment (JabobKrauskopf) [inline medmodels/medrecord/medrecord.py:788]:
```suggestion
                edges = [edges]
                
            edge_indices = self._medrecord.add_edges(edges)
```

**Review by JabobKrauskopf — CHANGES_REQUESTED** [review]

File: docs/user_guide/getstarted.md [hunk 9]
```diff
@@ -434,7 +422,7 @@ additional_young_id = medrecord.select_nodes(
     node().attribute("age").greater_or_equal(young_age)
     & node().attribute("age").less(higher_age)
 )
-medrecord.add_node_to_group(group="Young", node=additional_young_id)
+medrecord.add_nodes_to_group(group="Young", node=additional_young_id)
```
Comment (LauraBoenchenLB) [inline docs/user_guide/getstarted.md:425]:
```suggestion
medrecord.add_nodes_to_group(group="Young", nodes=additional_young_id)
```

File: docs/user_guide/getstarted.md [hunk 10]
```diff
@@ -446,12 +434,12 @@ print(
 It is possible to remove nodes from groups and to remove groups entirely from the MedRecord.
 
 ```python
-medrecord.remove_node_from_group(group="Young", node=additional_young_id)
+medrecord.remove_nodes_from_group(group="Young", node=additional_young_id)
```
Comment (LauraBoenchenLB) [inline docs/user_guide/getstarted.md:437]:
```suggestion
medrecord.remove_nodes_from_group(group="Young", nodes=additional_young_id)
```

**Review by LauraBoenchenLB — DISMISSED** [review]
Great Job! Everything is looking more uniform now 🦭

**Review by JabobKrauskopf — APPROVED** [review]
image

**Pull request merged**